src/error.rs

```diff
@@ -0,0 +1,49 @@
+use goblin::error::Error as GoblinError;
+use pdb_crate::Error as PDBError;
+pub type CompactSymbolResult<T> = Result<T, CompactSymbolError>;
+
+#[derive(Fail, Debug)]
+pub enum CompactSymbolError {
+    #[fail(display = "Strings do not match")]
+    UnmatchedString,
+
+    #[fail(display = "pdb_crate error: {}", _0)]
+    PDBError(PDBError),
+
+    #[fail(display = "Feature is not implemented")]
+    UnimplementedFeature,
+
+    #[fail(display = "IO Error")]
+    IOError,
+
+    #[fail(display = "Invalid string: {}", _0)]
+    InvalidStringError(&'static str),
+
+    #[fail(display = "Invalid result: {}", _0)]
+    InvalidResultError(&'static str),
+
+    #[fail(display = "None value error")]
+    NoneValueError,
+
+    #[fail(display = "goblin error: {}", _0)]
+    GoblinError(GoblinError),
+}
+
+impl From<PDBError> for CompactSymbolError {
+    fn from(err: PDBError) -> CompactSymbolError {
+        match err {
+            PDBError::UnimplementedFeature(_) => CompactSymbolError::UnimplementedFeature,
```

Comment:
For this one, I think it would be good to indicate that something was unimplemented in PDB specifically. So maybe it's easier to just take the default case for `PDBError::UnimplementedFeature` and simply wrap it in a `CompactSymbolError::PDBError`.